### PR TITLE
Group controls for flow_ebos

### PR DESCRIPTION
**Pull request by GitPaean**
Implementing the group controls for flow_ebos. VREP is a little special and possibly need to handle with other issues together, like OPM/opm-simulators#1011.  Will be proposed in another PR. 

This PR calculate the well potentials, and distribute the group target within different wells dynamically. GEFAC is also handled. The testing results are fine. 

This PR needs OPM/opm-core#1134 to get correct results, while not dependent on that PR in term of compiling.

### CODE REVIEW

**Comment by andlaus** [comment]
as far as I can tell the PR is good, but it I can't really review it because it is quite a bit outside of my comfort zone.  @totto82?

**Comment by GitPaean** [comment]
Please hold this PR and OPM/opm-core#1134  a little bit. In one time step it result in too small targets for all the wells in the group. I will investigate it tomorrow.

**Comment by GitPaean** [comment]
With updated OPM/opm-core#1134 and this PR, the result is recovered.  Now this PR is dependent on PR OPM/opm-core#1134 in term of compilation now. 

From my side, it is ready for review now.

**Comment by bska** [comment]
jenkins run norne spe polymer opm-core=1134 please

**Comment by atgeirr** [comment]
Missing is a test showing that flow_ebos still runs Norne and model2 correctly. Can you verify, @GitPaean?

File: opm/autodiff/StandardWellsDense.hpp [hunk 1]
```diff
@@ -1283,6 +1297,10 @@ enum WellVariablePositions {
 
                             break;
                         }
+
```
Comment (atgeirr) [inline opm/autodiff/StandardWellsDense.hpp]: Did you intend to insert some code here?

File: opm/autodiff/StandardWellsDense.hpp [hunk 2]
```diff
@@ -1474,16 +1512,170 @@ enum WellVariablePositions {
...
+                                // apply the strictest of the bhp controlls i.e. largest bhp for producers
+                                if (bhp_calculated > bhp) {
+                                    bhp = bhp_calculated;
+                                }
+                            } else {
+                                OPM_THROW(std::logic_error, "Expected PRODUCER or INJECTOR type of well");
+                            }
+                        }
+                    }
+
```
Comment (atgeirr) [inline opm/autodiff/StandardWellsDense.hpp]: At this point, I think bhp is expected to be nonzero, as there always must be a bhp limit, correct? If so, you should add an assert() to that effect.

File: opm/autodiff/StandardWellsDense.hpp (hunk 1 above)
Comment (GitPaean) [inline opm/autodiff/StandardWellsDense.hpp]: I wanted to add one line to separate these two parts. I will reduce the 4 blank lines to 1 line.

File: opm/autodiff/StandardWellsDense.hpp (hunk 2 above)
Comment (GitPaean) [inline opm/autodiff/StandardWellsDense.hpp]: I think it is reasonable. Will add an assert() here.

**Comment by atgeirr** [comment]
jenkins run norne norne_ebos spe polymer opm-core=1134 please

**Comment by atgeirr** [comment]
Reviewed and tested, will merge. Thanks for the effort!

**Pull request merged**